receiver/dockerstatsreceiver/README.md

```diff
@@ -0,0 +1,64 @@
+# Docker Stats Receiver
+
+The Docker Stats receiver queries the local Docker daemon's container stats API for
+all desired running containers on a configured interval.  These stats are for container
+resource usage of cpu, memory, network, and the
+[blkio controller](https://www.kernel.org/doc/Documentation/cgroup-v1/blkio-controller.txt).
+
+Requires Docker API version 1.22+.  At this time only Linux is supported.
+
+## Configuration
+
+The following settings are required:
+
+* `endpoint` (default = `unix:///var/run/docker.sock`): Address to reach the desired Docker daemon. 
+
+The following settings are optional:
+
+* `collection_interval`: (default = `10s`) The interval at which to gather container stats.
+* `timeout` (default = `5s`): The request timeout for any docker daemon query.
+* `container_labels_to_metric_labels` (no default): A map of Docker container label names whose label values to use
+as the specified metric label key.
+* `env_vars_to_metric_labels` (no default): A map of Docker container environment variables whose values to use
+as the specified metric label key.
+* `excluded_images` (no default, all running containers monitored): A list of strings,
+[regexes](https://golang.org/pkg/regexp/), or [globs](https://github.com/gobwas/glob) whose referent container image
+names will not be among the queried containers. `!`-prefixed negations are possible for all item types to signify that
+only unmatched container image names should be monitored.
+    * Regexes must be placed between `/` characters: `/my?egex/`.  Negations are to be outside the forward slashes:
+    `!/my?egex/` will monitor all containers whose name doesn't match the compiled regex `my?egex`.
+    * Globs are non-regex items (e.g. `/items/`) containing any of the following: `*[]{}?`.  Negations are supported:
+    `!my*container` will monitor all containers whose image name doesn't match the blob `my*container`.
+* `provide_all_block_io_metrics` (default = `false`): Whether to report metrics for all available blkio controller stats.
```

Comment:
I think given the architecture of the collector into pipelines with processors, the receiver should just emit everything and then let a processor drop stuff if desired.  There is no significant performance optimizations that can be made in this case by omitting certain metrics (other than a small bit of heap GC and copying values around when generating the MetricData).

Reply:
The `excludedImages` config does actually provide a performance optimization by avoiding API calls altogether so that is good.

Reply:
+1 excludedImages and I can remove the flags for blkio, memory, and network metrics.  I think limiting cpu by default is helpful because the per-core metrics have potentially relative cardinality concerns w/ few gains for general monitoring.

Reply:
Yeah could be useful for the per-core stuff given the extreme cardinality on big servers with dozens of virtual cores, but I would make the option explicit like `emit_per_core_metrics` or something to make it clear exactly what it controls.